### PR TITLE
skip multi-arch for fips-or-die test

### PR DESCRIPTION
**Pull request by jianzhangbjz**
Skip the `multi-arch` cluster test: https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.14-multi-nightly-azure-ipi-baselinecaps-v412-f28/1683896192812978176/artifacts/azure-ipi-baselinecaps-v412-f28/fips-check-fips-or-die/build-log.txt
```console
Pod v116-1 is Running.
Fail! the output of v116-1 is not as expected: exec ./app: exec format error
```

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-multi-nightly-azure-ipi-baselinecaps-v412-f28

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-azure-ipi-disconnected-f28

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-multi-nightly-openshift-logging-5.7-aws-ipi-ovn-ipsec-f7

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-multi-nightly-gcp-ipi-ovn-ipsec-f14

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh [hunk 1]
```diff
@@ -111,6 +111,22 @@ images=("quay.io/openshift-qe-optional-operators/myapp:v1.16-1"
...
+    auths=`cat /tmp/.dockerconfigjson`
+    if [[ $auths =~ "5000" ]]; then
+        echo "This is a disconnected env, skip it."
+        exit 0
+    fi
+fi
+
 node_name=`oc get node -l node-role.kubernetes.io/master= -o=jsonpath="{.items[0].metadata.name}"`
 arch=`oc get node/$node_name -o=jsonpath="{.status.nodeInfo.architecture}"`
 if [[ $arch != "amd64" ]]; then
```
Comment (aleskandro) [inline ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:132]: Beyond ARM64, are the other multi-arch folks aware of this skip?

/cc @Prashanth684 @andymcc @werled @prb112

**Comment by jianzhangbjz** [comment]
Hi @xiaojiey , could you help approve it? Thanks!

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh (hunk 1 above)
Comment (aleskandro) [reply ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:132]: /hold until reviews

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh [hunk 2]
```diff
@@ -111,6 +111,22 @@ images=("quay.io/openshift-qe-optional-operators/myapp:v1.16-1"
 set_proxy
 run_command "oc whoami"
 run_command "oc version -o yaml"
+
+version=`oc get clusterversion version -o=jsonpath="{.status.desired.version}"`
```
Comment (aleskandro) [inline ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:115]: Do not skip based on the clusterversion field.

Let's skip only if finding a node (worker or master) based on the arm64 architecture.

Also, what's blocking us from testing on P/Z, which images are not built for those architectures yet and where are their definitions and CI hosted?

If that's challenging to build for p/z yet, let's open a jira in the mulit-arch to track and discuss the images' build enablement and skip also if any node with p/z workers/masters is found.

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh [hunk 3]
```diff
@@ -111,6 +111,22 @@ images=("quay.io/openshift-qe-optional-operators/myapp:v1.16-1"
...
+
+run_command "oc extract secret/pull-secret -n openshift-config --confirm --to /tmp"; ret=$?
+if [[ $ret -eq 0 ]]; then
+    auths=`cat /tmp/.dockerconfigjson`
+    if [[ $auths =~ "5000" ]]; then
+        echo "This is a disconnected env, skip it."
+        exit 0
+    fi
+fi
+
```
Comment (aleskandro) [inline ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:129]: Remove the skip below on the CP's architecture in favour of the one above.

We should skip only if we find an arm64 node (either worker or master).

Starting at line 130

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh (hunk 1 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:132]: I think the `FIPS or die` test on `AMD64` is enough, @mrunalp What do you think? Thanks!

**Review by aleskandro — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh (hunk 1 above)
Comment (Prashanth684) [reply ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:132]: cc @jaypoulz FIPS mode is supported on P/Z 4.12+

**Review by jaypoulz — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh (hunk 2 above)
Comment (Prashanth684) [reply ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:115]: +1 - this field is not guaranteed  to be set. this is for the purpose of the user to indicate a desired transition to multi payload.

File: ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh (hunk 3 above)
Comment (Prashanth684) [reply ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:129]: it shold be both . if we are planning to support P/Z in the future  the condition here should be cp arch != worker arch then skip

Comment (aleskandro) [reply ci-operator/step-registry/fips-check/fips-or-die/fips-check-fips-or-die-commands.sh:129]: Yes, let me rephrase. Rather than executing two checks (1 for single-arch non-supported arches, another for multi), we can execute just one like in:
```
nodes=get_all_nodes()
for n in nodes:
  if  "$NON_SUPPORTED_ARCHES" == *n['metadata']['labels']['kubernetes.io/arch']*:
     echo "found node with arch {n['metadata']['labels']['kubernetes.io/arch']}. Skipping test"
     exit 0

```

this will be valid for both single-arch and multi clusters

**Comment by jianzhangbjz** [comment]
Close it since it has been replaced by #41952
